Actions/RunPipeline/RunPipeline.ps1

```diff
@@ -152,11 +152,30 @@ try {
         })
     }
 
-    $installApps = $settings.installApps
-    $installTestApps = $settings.installTestApps
+    $install = @{
+        "Apps" = $settings.installApps + @($installAppsJson | ConvertFrom-Json)
+        "TestApps" = $settings.installTestApps + @($installTestAppsJson | ConvertFrom-Json)
+    }
 
-    $installApps += $installAppsJson | ConvertFrom-Json
-    $installTestApps += $installTestAppsJson | ConvertFrom-Json
+    # Replace secret names in install.apps and install.testApps
+    foreach($list in @('Apps','TestApps')) {
+        $install."$list" = @($install."$list" | ForEach-Object {
+            $pattern = '.*(\$\{\{\s*([^}]+?)\s*\}\}).*'
+            if ($_ -match $pattern) {
+                $_.Replace($matches[1],[System.Text.Encoding]::UTF8.GetString([System.Convert]::FromBase64String($secrets."$($matches[2])")))
+            }
+            else {
+                $_
+            }
+            # Check validity of URL
+            try {
+                Invoke-WebRequest -Method Head -UseBasicParsing -Uri $_ | Out-Null
+            }
+            catch {
+                throw "Setting: install$($list) contains an inaccessible URL: $($_). Error was: $($_.Exception.Message)"
```

Comment:
At this point we've inserted the secret into `$_` so maybe we shouldn't throw an error with the full URL in it. I know GitHub will try to mask it but still... 
Maybe it'd be safer to log the original URL without the secret in it?

Reply:
Actually we haven't - we have written it to the output stream, why this also wouldn't work...